include/revng/Model/PrimitiveType.h

```diff
@@ -115,6 +115,60 @@ class model::PrimitiveType : public model::generated::PrimitiveType {
     return makeGeneric(model::Register::getSize(Register));
   }
 
+  /// @}
+
+public:
+  ///
+  /// \name Naming helpers
+  /// @{
+
+  std::string getCName() const {
+    if (PrimitiveKind() == model::PrimitiveKind::Void) {
+      revng_assert(Size() == 0);
+      return "void";
+    } else {
+      revng_assert(Size() != 0);
+      return model::PrimitiveKind::getCPrefix(PrimitiveKind()).str()
+             + std::to_string(Size() * 8) + "_t";
+    }
+  }
+
+  static std::string getCName(PrimitiveKind::Values Kind, uint64_t Size) {
+    return PrimitiveType(false, Kind, Size).getCName();
+  }
+
+  static UpcastableType fromCName(llvm::StringRef Name) {
+    // Figure the primitive kind out.
+    auto [Kind, RemainingName] = PrimitiveKind::tryConsumeCPrefix(Name);
+    if (Kind == PrimitiveKind::Invalid || RemainingName == Name)
+      return nullptr;
+
+    if (Kind == PrimitiveKind::Void)
+      return makeVoid();
+
+    // Ensure the name ends with _t
+    if (not RemainingName.consume_back("_t"))
+      return nullptr;
+
+    // Consume bit size
+    unsigned Bits = 0;
+    if (RemainingName.consumeInteger(10, Bits))
+      return nullptr;
+
+    // Ensure we consumed everything
+    if (RemainingName.size() != 0)
+      return nullptr;
+
+    // Ensure the bit size is a multiple of 8
+    if (Bits % 8 != 0)
+      return nullptr;
+
+    model::UpcastableType Result = make(Kind, Bits / 8);
+    revng_assert(Result->verify(true));
```

Comment:
Maybe instead of asserting we can return a PrimitiveType with invalid kind? So people can try to parse an type name, otherwise we'd need another method `isValidPrimitiveName()`.

Reply:
Done!